### PR TITLE
Tech 2701 integrate sagepay gateway

### PR DESCRIPTION
**Pull request by TomClockwork**

### CODE REVIEW

File: src/SagePay/Api/SagePayApi.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+class SagePayApi
+{
+  private $endpoint;
+  private $vendorName;
+  private $integrationKey;
+  private $integrationPassword;
+  public function __construct()
+  {
+    $this->endpoint = env('SAGE_PAY_BASE_URL');
```
Comment (GrantFS) [inline src/SagePay/Api/SagePayApi.php:22]: Please add default config e.g.

config/SagePay.php

File: src/SagePay/Api/SagePayApi.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      'paymentMethod' => [
+        'card' => [
+          'merchantSessionKey' => $merchantSessionKey->getMerchantSessionKey(),
+          'cardIdentifier' => $cardIdentifier->getCardIdentifier(),
+          'reusable' => false,
+          'save' => true,
+        ],
+      ],
+      'vendorTxCode' => uniqid(),
+      'amount' => 1,
```
Comment (GrantFS) [inline src/SagePay/Api/SagePayApi.php:79]: Price?

**Review by GrantFS — APPROVED** [review]

File: src/SagePay/Api/SagePayApi.php (hunk 1 above)
Comment (TomClockwork) [reply src/SagePay/Api/SagePayApi.php]: missing env variable already handled by Elite package, throws a `Clockwork\EliteParks\Exceptions\MissingAccessToken`

File: src/SagePay/Api/SagePayApi.php (hunk 2 above)
Comment (TomClockwork) [reply src/SagePay/Api/SagePayApi.php]: updated pr

Comment (TomClockwork) [reply src/SagePay/Api/SagePayApi.php]: updated

**Pull request merged**